src/common/types/chainner-scope.ts

```diff
@@ -64,10 +64,23 @@ struct NcnnNetwork {
 
 struct OnnxFile { path: string }
 struct OnnxModel {
-    scale: int(1..),
-    inputChannels: int(1..),
-    outputChannels: int(1..),
+    arch: string,
+    subType: string,
+    scaleHeight: int(1..) | null,
+    scaleWidth: int(1..) | null,
 }
+let OnnxRemBgModel = OnnxModel {
+    arch: "u2net",
+    subType: "RemBg",
+    scaleHeight: 1 | 3,
+    scaleWidth: 1,
+};
+let OnnxGenericModel = OnnxModel {
+    arch: invStrSet(OnnxRemBgModel.arch),
+    subType: "Generic",
+    scaleHeight: null,
+    scaleWidth: null,
```

Comment:
That's not really how types are supposed to be used. We don't need `null` to represent that we don't know the scale, because we just omit the field (all fields are optional).

Here's how I would do it:
```
struct OnnxModel {
    arch: string,
    subType: string,
    scaleWidth: int(1..),
    scaleHeight: int(1..),
}
let OnnxRemBgModel = OnnxModel {
    arch: "u2net",
    subType: "RemBg",
};
let OnnxGenericModel = OnnxModel {
    arch: invStrSet(OnnxRemBgModel.arch),
    subType: "Generic",
}
```

I mean, generic models do have a scale, we just don't have a way of knowing it yet.

Reply:
Why remove the specific scale definitions from OnnxRemBgModel?

Reply:
Because (1) they will be "added back in" when the model is loaded and gets its output type overridden and (2) because it's an implementation detail (if there is a model with a height scale of 2, we'd have to change the type again).